classes/order/OrderHistory.php

```diff
@@ -615,4 +627,47 @@ public function addWs()
             return $this->add();
         }
     }
+
+    /**
+     * @return MailPartialTemplateRenderer
+     */
+    protected function getPartialRenderer()
+    {
+        if (!$this->partialRenderer) {
+            $this->partialRenderer = new MailPartialTemplateRenderer($this->context->smarty);
+        }
+
+        return $this->partialRenderer;
+    }
+
+    /**
+     * Fetch the content of $template_name inside the folder
+     * current_theme/mails/current_iso_lang/ if found, otherwise in
+     * mails/current_iso_lang.
+     *
+     * @param string $template_name template name with extension
+     * @param int $mail_type Mail::TYPE_HTML or Mail::TYPE_TEXT
+     * @param array $var sent to smarty as 'list'
+     *
+     * @return string
+     */
+    protected function getEmailTemplateContent($template_name, $mail_type, $var)
+    {
+        $email_configuration = Configuration::get('PS_MAIL_TYPE');
+        if ($email_configuration != $mail_type && $email_configuration != Mail::TYPE_BOTH) {
+            return '';
+        }
+
+        if (!isset($this->context)) {
+            $this->context = Context::getContext();
+        }
+
+        $content = $this->getPartialRenderer()->render($template_name, $this->context->language, $var);
+        if ($mail_type == Mail::TYPE_TEXT) {
+            // delete html comments in text mode so it's not visible in the mail
+            $content = preg_replace('/\n?<!--.*-->\n?/i', '', $content);
```

Comment:
Maybe any spaces types? Don't know if all templates contains only `\n` notations, or if a developer add an email created on Mac / Windows.
Add global and dot can match new lines with ungreedy notation
for this kind of comments:
```html
<!-- blyat -->
<p>Htmlcontent</p>
<!-- 
test 
-->
```

```suggestion
            $content = preg_replace('/\s?<!--.*?-->\s?/gs', '', $content);
```

Reply:
I did that mainly because the partial renderer add the partial file being rendered in html comment like that : 
```
return "\n<!-- begin $tpl -->\n"
            . parent::fetch($template, $cache_id, $compile_id, $parent, $display, $merge_tpl_vars, $no_output_filter)
            . "\n<!-- end $tpl -->\n";
```

Reply:
Oh, I was more thinking about comments in custom html template ^^'

Reply:
I don't understand the need to remove html comment tags, when I look at your txt template `download_product_virtual_products.txt` there is no html content, so why do you need to remove it?

Reply:
That's because when  the `download_product_virtual_products.txt` template is rendered, the renderer adds comments. Here is what I get in my mailbox when I don't remove html comments : 
```
http://localhost/prestashop/index.php

Hi thomas baccelli,

Thank you for your order with the reference JRNKCBIHJ from PrestaShop

Product(s) to download

You have 1 product(s) now available for download using the following link(s):


<!-- begin /Users/tBaccelli/Sites/prestashop/mails/_partials/download_product_virtual_products.txt -->
  [Capture d&rsquo;écran 2019-10-15 à 17.40.05.png](http://localhost/prestashop/index.php?controller=get-file&key=756cddd141c35f16ab816bb301b14b79de34182f-6d13e7f3990af715aeb737adb923a46b5ee456fa&id_order=1932&secure_key=24926c1a734a9822e6ad4f535668f657)

<!-- end /Users/tBaccelli/Sites/prestashop/mails/_partials/download_product_virtual_products.txt -->


Follow your order and download your invoice on our shop, go to the [Order history and details](http://localhost/prestashop/index.php?controller=history) section of your customer account.

If you have a guest account, you can follow your order via the [Guest Tracking](http://localhost/prestashop/index.php?controller=guest-tracking) section on our shop.

[PrestaShop](http://localhost/prestashop/index.php)

Powered by [PrestaShop](https://www.prestashop.com/?utm_source=marchandprestashop&utm_medium=e-mail=utm_campaign=footer_1-7)

```

Reply:
Oh damn, seems like it's coming from our own over layer on Smarty (probably SmartyDev in this case) https://github.com/PrestaShop/PrestaShop/blob/4ec595f06d52c5aa93b8dd930699f8f6d21cca58/classes/Smarty/SmartyDev.php#L39

This is completely not needed in this case, so we shouldn't use the context Smarty instance but a normal one instead.. but the thing is smarty is configured in https://github.com/PrestaShop/PrestaShop/blob/develop/config/smarty.config.inc.php

Then maybe we should add a `cleanComments` parameter to the `MailPartialTemplateRenderer::render` method